BTCPayServer/HostedServices/InvoiceWatcher.cs

```diff
@@ -321,6 +323,8 @@ private async Task<bool> UpdateConfirmationCount(InvoiceEntity invoice)
                         {
                             var transactionResult = await _ExplorerClientProvider.GetExplorerClient(payment.GetCryptoCode())?.GetTransactionAsync(onChainPaymentData.Outpoint.Hash);
                             var confirmationCount = transactionResult?.Confirmations ?? 0;
+                            if (onChainPaymentData.Network.CryptoCode == "DASH" && confirmationCount == 0 && transactionResult != null)
+                                confirmationCount = await GetDashInsightExplorerTxLock(onChainPaymentData.Outpoint.Hash) ? 6 : 0;
```

Comment:
It's maybe just a misunderstanding on my part, but it looks like you set the `confirmationCount` to 6 if the InstantSend is confirmed by Insight Explorer.

Later in the `BTCPayServer/Views/Stores/UpdateStore.cshtml` file you changed the text for the 1 confirmation case (not the 6). 

Was this on purpose?

Reply:
InstantSend means the transaction is instantly locked in by masternodes to be treated as 6 confirmations. Now with ChainLocks since 0.14 this doesn't make much sense anymore as any transaction (including all InstantSend transactions) are completely confirmed and can never be changed again once the ChainLock for the block is set, which is much safer anyway (no chain reorg can happen).

The whole concept of confirmations makes just less sense for DASH, you either have it unconfirmed (less than 1% of normal transactions) or it is somehow already confirmed and as a user or merchant you usually don't worry about it and can be sure you have received your money within 1-2s.

The actual problem was that no explorer was returning txlock correctly for the past months, we are in a pretty good situation now that it works so well and we can simply ask any insight explorer and it will give us confirmation quite quickly that a transaction has been locked in. Last I tested about 1.5 months ago this didn't work on any explorer at all.

Reply:
plus NBXplorer doesn't properly support any of the dash advanced features of 0.13 or 0.14 and I was to lazy to change that as well, we tried using NBXplorer, but kept using insight explorer as our usecase is mostly directly from javascript and we didn't touch our backend services much this year (which are mostly in c#, thus NBXplorer is a good choice there).

Reply:
Thank you for the detailed explanation, I really appreciate it!

In this case wouldn't it be more accurate to change the [BTCPayServer/Views/Stores/UpdateStore.cshtml](https://github.com/snogcel/btcpayserver/pull/1/files#diff-b3d609ddf365d04e5e5062ebec2a91b9) file, so the `(or is InstantSend confirmed)` is in the 6 confirmations line?